internal/upload/koji/koji.go

```diff
@@ -304,16 +309,47 @@ func (k *Koji) uploadChunk(chunk []byte, filepath, filename string, offset uint6
 	q.Add("fileverify", "adler32")
 	u.RawQuery = q.Encode()
 
-	client := http.Client{Transport: k.transport}
+	retries := uint(0)
+
+	countingCheckRetry := func(ctx context.Context, resp *http.Response, err error) (bool, error) {
+		retries++
+		shouldRetry, err := retryablehttp.DefaultRetryPolicy(ctx, resp, err)
+
+		if !shouldRetry && err != nil {
```

Comment:
These 3 conditions are non-trivial, can you please add a comment describing the decision logic?

Reply:
I have added some comments to explain what's going on.
I can add more, let me know what you think. The conditions are the same as in DefaultRetryPolicy (except the first one), both trying to make sure it's the right error before taking a decision to retry or not.